pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -155,15 +156,22 @@ type reconciler struct {
 
 	// idAllocator provides interfaces to allocate and release uint32 id.
 	idAllocator *idAllocator
+
+	// priorityAssigner provides interfaces to manage OF priorities.
+	priorityAssigner *priorityAssigner
+
+	// priorityMutex prevents concurrent priority re-assignments
+	priorityMutex sync.RWMutex
```

Comment:
Could this be priorityAssigner internal?

Reply:
No it cannot. We need to treat priority-reassignments (including on OVS table) caused by  priority insertion as atomic operation. So the mutex needs to span across reconciler.getOFPriority() and reconciler.installOFRule(). Putting the mutex in priorityAssigner will break this atomicity.

Reply:
I don't think `conjMatchFlowLock` alone works here. We need a mutex at reconciler level